Pidaq/Lib/database/database.py

```diff
@@ -0,0 +1,194 @@
+        self.client.switch_database(database)
+        self.current_database = database
+
+    def log_data(self, data: dict, measurement: str, tags: dict) -> None:
+        """Logs a single data point to the database"""
+
+        table_row = [{
+            'measurement': measurement,
+            'time': datetime.datetime.now(),
+            'fields': data,
+        }]
+        self.client.write_points(table_row, tags=tags)
+
+    def read_data(self, tags=None, measurements=None) -> str:
+        """Reads database for specified measurements, including specified tags and fields
+
+        Args:
+            tags (list(str)): Contains the tags and fields to be included in the query
+            measurements (list(str)): Contains the measurements to be queried
+
+        Returns:
+            (str) Raw json data
+        """
+
+        if tags is not None and measurements is None:
+            formatted_tags = format_data(tags)
+            data = self.client.query(
+                f'SELECT {formatted_tags}',
+                database=self.current_database
+            )
+        elif tags is None and measurements is not None:
+            formatted_measurements = format_data(measurements)
+            data = self.client.query(
+                f'SELECT * FROM {formatted_measurements}',
+                database=self.current_database
+            )
+        elif tags is not None and measurements is not None:
+            formatted_tags = format_data(tags)
+            formatted_measurements = format_data(measurements)
+            data = self.client.query(
+                f'SELECT {formatted_tags} FROM {formatted_measurements}',
+                database=self.current_database
+            )
+        else:
+            return -1 # Must specify at least a tag or a measurement
+        try:
+            return data.raw['series']
+        except KeyError:
+            return -1
+
+    def create_retention_policy(self, name: str, duration: str, replication: str) -> None:
+        """Creates a retention policy for the current database"""
+
+        self.client.create_retention_policy(
+            name,
+            duration,
+            replication,
+            database=self.current_database,
+            default=True
+        )
+
+    def export_to_csv(self, test_name: str, tags=None, measurements=None, csv_path=None) -> None:
+        """Exports measurements to a csv file
+
+        Args:
+            test_name (str): Name of test, becomes name of csv file
+            tags (list(str)): Tags and fields to be included in csv
+            measurements (list(str)): Measurements to be included in csv
+            csv_path(str): File path of where the csv will be written
+        """
+
+        file_name = f'{test_name}.csv'
+        data = self.read_data(tags=tags, measurements=measurements)
+        if csv_path is None:
+            csv_path = os.getcwd()
+        with open(f'{csv_path}\\{file_name}', 'w', newline='\n') as csvfile:
```

Comment:
Im thinking this could be a little more cross platform using os.path.join(). The current path implementation seems Windows-oriented ("I think?")

`with open(os.path.join(csv_path, file_name))...` would resolve the path based on the host OS.

 Source - Denis Rouleau, Member of Technical Staff at Wind River Systems.

Reply:
Also, maybe we could add a date to the file name - nothing long or verbose just a simple indication of the date.

Also Also, If we could prepend the file with some metadata that would also be good. Not sure if thats a good idea though because that goes against the csv format. 

Just thinking outloud- maybe a metadata file with the same name that gets created here? It could have the following fields: 
`test profile name` - easy
`date` - easy
`operator name` - this would be a neat feature, simple input box on the gui
`commands` - this would also be simple. Just accept a list of something (dict, string, whatever) that contains each command executed. This export will get called somewhere where that info is readily available anyways

What do you guys think?

Reply:
Ah yes, good idea with the `os.path.join`, I was thinking about that while writing and forgot to figure out how to do it. Will make that change.

I agree that this is a good idea, and creating a separate file for it makes sense cause of what you mentioned about csv formatting, it will be tidier to have a separate file.

Reply:
I am gonna have a separate function for creating a metadata file, I don't want there to be too much going on in this method

Reply:
Good idea, may be good for that function to take a list of commands that the motor was sent and the test name